QPM/SymLinker.Linker/Linker.cs

```diff
@@ -0,0 +1,269 @@
+﻿using SymLinker.Linker.LinkCreators;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+
+// Stolen from https://github.com/huffSamuel/SymLinker
+// Cleaned up by Fern, because the old code was horrible
+namespace SymLinker.Linker
+{
+    public class Linker
+    {
+        #region Actions
+        // Attach to these to get warnings, errors, and informaion out of SymLinkerCore
+        public Action<string> OnWarn;
+        public Action<string> OnError;
+        public Action<string> OnInfo;
+        #endregion
+
+        private List<Error> errors = new List<Error>();
+        private ISymLinkCreator linker = null;
+
+        /// <summary>
+        /// Creates a new Symbolic Link Creator
+        /// </summary>
+        public Linker()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                linker = new WindowsSymLinkCreator();
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                linker = new LinuxSymLinkCreator();
+            }
+            else
+            {
+                linker = new OSXSymLinkCreator();
+            }
+        }
+
+        /// <summary>
+        /// Creates a symbolic link from <paramref name="source"/> to <paramref name="dest"/>
+        /// </summary>
+        /// <param name="source">Source file</param>
+        /// <param name="dest">Destination directory</param>
+        /// <returns>
+        /// Returns true if the system was able to create the SymLink
+        /// </returns>
+        public void CreateLink(string source, string dest)
```

Comment:
Perhaps the thing I like least about this code structure is that if `CreateLink` is _going_ to fail because of it not being supported on Linux/OSX, it still calls into this and performs a `CheckLinkReadiness` _and_ writes any errors.

This could be solved by moving around your specific checks, specifically checking for runtime validity before attempting to do anything. This issue is a fairly obnoxious one and I'd rather avoid throwing exceptions simply for the sake of throwing exceptions, instead handling non-windows OSes by properly exiting early.